packages/cli/lib/artifact-generator.js

```diff
@@ -93,27 +93,7 @@ module.exports = class ArtifactGenerator extends BaseGenerator {
     });
 
     if (generationStatus) {
-      /**
-       * Update the index.ts in this.artifactInfo.outDir. Creates it if it
-       * doesn't exist.
-       * this.artifactInfo.outFile is what is exported from the file.
-       *
-       * Both those properties must be present for this to happen. Optionally,
-       * this can be disabled even if the properties are present by setting:
-       * this.artifactInfo.disableIndexUpdate = true;
-       */
-      if (
-        this.artifactInfo.outDir &&
-        this.artifactInfo.outFile &&
-        !this.artifactInfo.disableIndexUpdate
-      ) {
-        await updateIndex(this.artifactInfo.outDir, this.artifactInfo.outFile);
-        // Output for users
-        this.log(
-          chalk.green('   update'),
-          `${this.artifactInfo.relPath}/index.ts`,
-        );
-      }
+      await this._updateIndexFiles();
```

Comment:
Should `updateIndexFiles` be a step before `end`?

Reply:
It's part of `end` at the moment.

Reply:
See http://yeoman.io/authoring/running-context.html

Reply:
Moving it would mean also moving/duplication the `generationStatus` check.